analytics_data_api/v0/serializers.py

```diff
@@ -342,10 +340,17 @@ def get_engagements(self, obj):
         Add the engagement totals.
         """
         engagements = {}
-        for entity_type in engagement_entity_types.AGGREGATE_TYPES:
-            for event in engagement_events.EVENTS[entity_type]:
-                metric = '{0}_{1}'.format(entity_type, event)
-                engagements[metric] = getattr(obj, metric, 0)
+
+        # fill in these fields will 0 if values not returned/found
+        default_if_none_fields = ['discussions_contributed', 'problems_attempted',
+                                  'problems_completed', 'videos_viewed']
+        for field in default_if_none_fields:
+            engagements[field] = self.default_if_none(getattr(obj, field, None), 0)
+
+        # preserve null values for problem attempts per completed
+        no_default_field = 'problem_attempts_per_completed'
```

Comment:
This is a nitpick, but since this code path only applies to this field, I think the variable name could be a little less general.